packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -102,6 +136,129 @@ function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
       oid: pd.gateway.oid,
       url,
       id: pd.gateway.id,
+      ...(pd.patientMatch ? { patientResource: pd.patientResource } : undefined),
     };
   });
 }
+
+export async function runNextPdOnNewDemographics({
+  patient,
+  facilityId,
+  requestId,
+  cqLinks,
+}: {
+  patient: Patient;
+  facilityId: string;
+  requestId: string;
+  cqLinks: CQLink[];
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const newDemographicsHere = getNewDemographics(updatedPatient, cqLinks);
+  const foundNewDemographicsHere = newDemographicsHere.length > 0;
+  const foundNewDemographicsAcrossHies = await checkLinkDemographicsAcrossHies({
+    patient: updatedPatient,
+    requestId,
+  });
+  if (foundNewDemographicsHere || foundNewDemographicsAcrossHies) {
+    if (foundNewDemographicsHere) {
+      await updatePatientLinkDemographics({
+        requestId,
+        patient: updatedPatient,
+        source: MedicalDataSource.CAREQUALITY,
+        links: newDemographicsHere,
+      });
+    }
+    await discover({
+      patient: updatedPatient,
+      facilityId,
+      rerunPdOnNewDemographics: false,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.rerunOnNewDemographics,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        foundNewDemographicsHere,
+        foundNewDemographicsAcrossHies,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function runNexPdIfScheduled({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
```

Comment:
Why are we querying the patient again?

Reply:
In case there was a scheduled PD in the interim.

Reply:
Sounds good, makes sense!